### PR TITLE
Fix: unable to read configuration file when path contains Chinese

### LINKED ISSUE #9957: Under the environment of Ubuntu + eclipse, the path of project directory contains Chinese, which leads to the error that shardingsphere proxy cannot read the configuration file
Description: Under the environment of Ubuntu + eclipse, the path of project directory contains Chinese, which leads to the error that shardingsphere proxy cannot read the configuration file

Version: shardingsphere:5.0.0-alpha

Environment: Ubuntu:20 + Eclipse Oxygen.3a Release (4.7.3a)

Errors Stacktrace:

Exception in thread "main" java.io.FileNotFoundException: /home/xxx/%e6%96%87%e6%a1%a3/java-workspace/shardingsphere/shardingsphere-proxy/shardingsphere-proxy-bootstrap/target/classes/conf/server.yaml (没有那个文件或目录)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at org.apache.shardingsphere.infra.yaml.engine.YamlEngine.unmarshal(YamlEngine.java:55)
	at org.apache.shardingsphere.proxy.config.ProxyConfigurationLoader.loadServerConfiguration(ProxyConfigurationLoader.java:77)
	at org.apache.shardingsphere.proxy.config.ProxyConfigurationLoader.load(ProxyConfigurationLoader.java:60)
	at org.apache.shardingsphere.proxy.Bootstrap.main(Bootstrap.java:47)

### PR DESCRIPTION
**Pull request by wl4g**
Fixes #9957.

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-common/src/main/java/org/apache/shardingsphere/proxy/config/ProxyConfigurationLoader.java [hunk 1]
```diff
@@ -64,9 +67,14 @@ public static YamlProxyConfiguration load(final String path) throws IOException
                 YamlProxyRuleConfiguration::getSchemaName, each -> each, (oldValue, currentValue) -> oldValue, LinkedHashMap::new)));
     }
     
-    private static File getResourceFile(final String path) {
+    private static File getResourceFile(final String path) throws IOException {
         URL url = ProxyConfigurationLoader.class.getResource(path);
-        return null == url ? new File(path) : new File(url.getFile());
+        if (null != url) {
+            // Fixed: the path contains Chinese yaml error.
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-common/src/main/java/org/apache/shardingsphere/proxy/config/ProxyConfigurationLoader.java:73]: Could remove the commit? Codes are the best explanation.

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-common/src/main/java/org/apache/shardingsphere/proxy/config/ProxyConfigurationLoader.java (hunk 1 above)
Comment (wl4g) [reply shardingsphere-proxy/shardingsphere-proxy-common/src/main/java/org/apache/shardingsphere/proxy/config/ProxyConfigurationLoader.java:73]: I agree.

**Review by terrymanu — APPROVED** [review]

**Review by terrymanu — CHANGES_REQUESTED** [review]
Please resolve conflict

File: shardingsphere-proxy/shardingsphere-proxy-common/src/main/java/org/apache/shardingsphere/proxy/config/ProxyConfigurationLoader.java (hunk 1 above)
Comment (wl4g) [reply shardingsphere-proxy/shardingsphere-proxy-common/src/main/java/org/apache/shardingsphere/proxy/config/ProxyConfigurationLoader.java]: Conflict resolved!

**Review by terrymanu — CHANGES_REQUESTED** [review]
The CI is broken, could you fix it?

File: shardingsphere-proxy/shardingsphere-proxy-common/src/main/java/org/apache/shardingsphere/proxy/config/ProxyConfigurationLoader.java [hunk 2]
```diff
@@ -64,9 +67,12 @@ public static YamlProxyConfiguration load(final String path) throws IOException
                 YamlProxyRuleConfiguration::getSchemaName, each -> each, (oldValue, currentValue) -> oldValue, LinkedHashMap::new)));
     }
     
-    private static File getResourceFile(final String path) {
+    private static File getResourceFile(final String path) throws IOException {
         URL url = ProxyConfigurationLoader.class.getResource(path);
-        return null == url ? new File(path) : new File(url.getFile());
+        if (null != url) {
```
Comment (kimmking) [inline shardingsphere-proxy/shardingsphere-proxy-common/src/main/java/org/apache/shardingsphere/proxy/config/ProxyConfigurationLoader.java:72]: return new File(URLDecoder.decode(null != url ? url.getFile() : path, "UTF-8"));

will be better.

**Review by kimmking — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-common/src/main/java/org/apache/shardingsphere/proxy/config/ProxyConfigurationLoader.java (hunk 2 above)
Comment (wl4g) [reply shardingsphere-proxy/shardingsphere-proxy-common/src/main/java/org/apache/shardingsphere/proxy/config/ProxyConfigurationLoader.java:72]: I agree

**Comment by tristaZero** [comment]
Hi @wl4g Please have a look here, if there is no feedback, we will close it after 3 days. Thanks.

**Comment by tristaZero** [comment]
Hi I have to close this one since there is no feedback anymore.

**Pull request closed without merging**